fftvis/simulate.py

```diff
@@ -149,6 +152,15 @@ def simulate_vis(
     if not polarized:
         beam = prepare_beam_unpolarized(beam, use_feed=use_feed)
 
+    # Sky model should either be unpolarized or have shape (nsources, nfreqs, 4)
+    #if not (
+    #    (polarized and len(fluxes.shape) == 3 and fluxes.shape[-1] != 4) or 
+    #    (polarized and len(fluxes.shape) != 2)
+    #): 
+    #    raise ValueError(
+    #        "If polarized is True, fluxes must have shape (4, nsources, nfreqs) or (nsources, nfreqs)."
+    #    )
+
```

Comment:
If you do this check inside the `stokes_to_coherency` function, you can just remove this commented code